src/init.cpp

```diff
@@ -1968,8 +1968,10 @@ bool AppInit2(boost::thread_group& threadGroup, CScheduler& scheduler)
     mnodeman.UpdatedBlockTip(chainActive.Tip());
     privateSendClient.UpdatedBlockTip(chainActive.Tip());
     mnpayments.UpdatedBlockTip(chainActive.Tip());
-    masternodeSync.UpdatedBlockTip(chainActive.Tip());
+    masternodeSync.UpdatedBlockTip(chainActive.Tip(), true);
     governance.UpdatedBlockTip(chainActive.Tip());
+    // make sure to skip IBD state in sync if we are already up to date
+    masternodeSync.UpdatedBlockTip(chainActive.Tip(), IsInitialBlockDownload());
```

Comment:
UpdatedBlockTip logic for MN Sync is confusing, even with the comment.

Why is `masternodeSync.UpdatedBlockTip` called a second time?
Why not straight after the first call?
Can they be combined into one call?

The logic is mostly hidden in the function and not obvious from the function name
(top level UpdatedBlockTip is a callback from validationinterface, these should be simple helpers?).
I'd suggest to move as much logic as possible out of the function - the external logic can then call SwitchToNextAsset twice if required "to skip IBD".

Here's the definition with expanded logic for reference:
```
void CMasternodeSync::UpdatedBlockTip(const CBlockIndex *pindexNew, bool fInitialDownload)
{
   pCurrentBlockIndex = pindexNew;
   if((fInitialDownload && nRequestedMasternodeAssets == MASTERNODE_SYNC_INITIAL) ||
     (!fInitialDownload && nRequestedMasternodeAssets <= MASTERNODE_SYNC_IBD))
   {
       SwitchToNextAsset();
   }
   if((nRequestedMasternodeAssets != MASTERNODE_SYNC_FINISHED) &&
      (nRequestedMasternodeAssets != MASTERNODE_SYNC_FAILED))
           nTimeLastBumped = GetTime(); // postpone timeout because of new block
 }
```
1st call with `fInitialDownload=true` does SwitchToNextAsset only from INITIAL (switching to IBD)
(then bumps time if not FINISHED or FAILED).
2nd call with `IsInitialBlockDownload()==true` does nothing but bump time again. 
2nd call with `IsInitialBlockDownload()==false` does SwtichToNextAsset from INITIAL (to IBD), IBD (to LIST) or FAILED (runtime error)
(then bumps time if not FINISHED or FAILED).